MNTOR-1509 - re-implement DNT check so it works with Safari

rhelmer opened the pull request:
<!-- The following is intended to be helpful to you. Feel free to remove anything that is not. -->

# References: 
Jira: MNTOR-1509


<!-- When adding a new feature: -->

# Description

The Do Not Track (DNT) header is deprecated and not implemented in all browsers (notably Safari), but until we have a better opt-out mechanism we should keep this in place to be consistent with other Mozilla sites.

rhelmer commented on src/views/guestLayout.js:
```diff
@@ -42,28 +42,33 @@ const guestLayout = data => `
+        'https://www.googletagmanager.com/gtag/js?id='+i+dl;var n=d.querySelector('[nonce]');
+        n&&j.setAttribute('nonce',n.nonce||n.getAttribute('nonce'));f.parentNode.insertBefore(j,f);
+        })(window,document,'script','dataLayer','${AppConstants.GA4_MEASUREMENT_ID}');
+        function gtag(){dataLayer.push(arguments);}
```
This big blob of code is what's recommended on https://developers.google.com/tag-platform/tag-manager/web/csp to work with script nonce FWIW :)

mansaj replied:
So is this a revert of the previous PR or a little bit more?

rhelmer replied:
It's slightly different, it checks for `navigator.doNotTrack` explicitly now. Safari doesn't have that API anymore, and the way it was written before it disabled GA erroneously.

mansaj reviewed (approved):
Looks to me like a revert of the previous PR

rhelmer commented:
> Looks to me like a revert of the previous PR

It is but I also corrected the check so it works correctly if `navigator.doNotTrack` is undefined (Safari doesn't implement it), and cleaned up the config section as well.

toufali commented on src/views/guestLayout.js:
```diff
@@ -42,28 +42,32 @@ const guestLayout = data => `
-        { cookie_domain: window.location.hostname, cookie_flags: "SameSite=None;Secure" }
-      );
-      window.gtag = gtag
+      if (!navigator.doNotTrack || navigator.doNotTrack !== '1') {
```
Shouldn't `navigator.doNotTrack !== '1'` also cover the case when navigator.doNotTrack is undefined?

toufali commented on src/views/guestLayout.js:
```diff
@@ -42,28 +42,32 @@ const guestLayout = data => `
-        { cookie_domain: window.location.hostname, cookie_flags: "SameSite=None;Secure" }
-      );
-      window.gtag = gtag
+      if (!navigator.doNotTrack || navigator.doNotTrack !== '1') {
```
Shouldn't navigator.doNotTrack !== '1' also cover the case when navigator.doNotTrack is undefined?

rhelmer replied:
Hm, yeah for some reason I thought it would throw there, but you're right. I'll clean this up separately, thanks!
